x-pack/legacy/plugins/code/index.ts

```diff
@@ -93,6 +93,15 @@ export const code = (kibana: any) =>
             .default(['https', 'git', 'ssh']),
           enableGitCertCheck: Joi.boolean().default(true),
         }).default(),
+        disk: Joi.object({
```

Comment:
Trying to applying what ES does as documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/disk-allocator.html). Will have a scheduler to check disk spaces repeatedly and persist some flag so that clone/update will depends on it.

Or implement a simpler version that everytime a clone/update happens, check the disk usage on the fly and only use one watermark threshold instead of 3.

@zfy0701 thoughts?

Reply:
I am leaning toward the later simplified one.

Reply:
I agree with the simpler solution as well